kubernetes/resource_kubernetes_certificate_signing_request.go

```diff
@@ -0,0 +1,208 @@
+package kubernetes
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"k8s.io/api/certificates/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"log"
+	"reflect"
+	"time"
+)
+
+func resourceKubernetesCertificateSigningRequest() *schema.Resource {
+	apiDoc := v1beta1.CertificateSigningRequest{}.SwaggerDoc()
+	apiDocSpec := v1beta1.CertificateSigningRequestSpec{}.SwaggerDoc()
+	apiDocStatus := v1beta1.CertificateSigningRequestStatus{}.SwaggerDoc()
+
+	return &schema.Resource{
+		Create: resourceKubernetesCertificateSigningRequestCreate,
+		Read:   resourceKubernetesCertificateSigningRequestRead,
+		Delete: resourceKubernetesCertificateSigningRequestDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"auto_approve": {
+				Type: schema.TypeBool,
+				Description: "By choosing this option, Terraform will automatically approve the CertificateSigningRequest" +
+					" and a certificate will be issued. Note that this will save the certificate in Terraform's state file, " +
+					"which makes the certificate readable in plaintext on the system where Terraform was run. " +
+					"It is recommended to use a remote backend when selecting this option, to prevent local users " +
+					"from reading the certificate data. https://www.terraform.io/docs/state/remote.html",
```

Comment:
@jrhouston I'm going to remove the warning about saving the certificate in state. Does that sound ok? Kubernetes only generates a public cert, which is not sensitive data (it is freely given to remote servers by design). I originally had this warning here because at a previous employer, they liked to keep the certificates hidden.